plugins/javadoc/src/main/kotlin/org/jetbrains/dokka/javadoc/pages/utils.kt

```diff
@@ -2,14 +2,20 @@ package org.jetbrains.dokka.javadoc.pages
 
 import org.jetbrains.dokka.model.*
 
-internal fun JavadocFunctionNode.getAnchor(): String =
-    "$name(${parameters.joinToString(",") {
-        when (val bound = if (it.typeBound is Nullable) it.typeBound.inner else it.typeBound) {
-            is TypeConstructor -> listOf(bound.dri.packageName, bound.dri.classNames).joinToString(".")
-            is TypeParameter -> bound.name
-            is PrimitiveJavaType -> bound.name
-            is UnresolvedBound -> bound.name
-            is JavaObject -> "Object"
-            else -> bound.toString()
-        }
-    }})"
\ No newline at end of file
+internal fun JavadocFunctionNode.getAnchor(): String {
+    val parameters = parameters.joinToString(",") { it.typeBound.asString() }
+    return "$name($parameters)"
+}
+
+private fun Bound.asString(): String = when (this) {
+    is Nullable -> this.inner.asString()
+    is DefinitelyNonNullable -> this.inner.asString()
+    is TypeConstructor -> listOf(this.dri.packageName, this.dri.classNames).joinToString(".")
+    is TypeParameter -> this.name
+    is PrimitiveJavaType -> this.name
+    is UnresolvedBound -> this.name
+    is TypeAliased -> this.typeAlias.asString()
```

Comment:
`TypeAliased` was missing, so it defaulted to `toString()`, which leaked all configuration details.

Reply:
You also *changed* the behavior of Dynamic and Void, those need tests to lock in the "correct" behavior?
```
object Void : Bound()
object Dynamic : Bound()
```
`bound.toString()` was `Void@hashcode`, but now it's `void`.

Reply:
Currently, under the normal circumstances, it is not possible to get `Dynamic` or `Void` bounds in the Javadoc format. 

`Dynamic` is JS specific, and the Javadoc format does not support multiplatform projects (it has a source set checker), while the `Void` Bound is only used as the return type. If you accept `Void` as a function parameter, it'll be `TypeConstructor` instead (so `java.lang.Void`). For that reason, it's not possible to add proper tests

Yet it feels a bit scary to be throwing the exception in these cases, as the checker can be disabled without much trouble, and we or someone else might add support for multiplatform projects or another mapping for the `Void` bound, so they're still handled as they should. I've added comments to explain it though

I keep a list of similar "assumptions" that Dokka makes to play it safe, and one day I'll get to using asserts that only work in debug/dev builds of Dokka, so that we can catch these things. That is, for `Void` it would be something like `if (isInDevMode) error("Add proper support") else "void"`

Added a test for definitely non nullable types :+1: